### DISCUSSION TITLE
dynamic config

### DISCUSSION THREAD

**Discussion by bulldog98** [quarkusio/quarkus#34333] [Q&A]
If I understood the smallrye-config correctly changes in e.g. the application.properties should be reflected directly.

Given some config `SomeConfig` how would I derive an other type from that, that is always reflective of the config.

Example I have `SomeConfig` but need `DerivedType` for some library I need to use. I've also have a converter function `derive`, how would I always get the latest `DerivedType`, do I always have to call `derive`? My expectation would be that I'm able to inject a `Uni<SomeConfig>` and be able to derive from that, since creating object 1000s of times a second is, as far as I know, expensive.

**Comment by gsmet** [comment]
Could you please give a concrete example with code of what you are trying to achieve and what doesn't work?

**Reply by bulldog98** [reply]
@gsmet 
```java
@ConfigMapping(prefix = "any")
interface SomeConfig {
    int port();
    String host();
}

@Singleton
class SomeClass {
    DerivedType derivedType;

    SomeClass(SomeConfig config) {
        derivedType = new DerivedType(config.port(), config.host());
    }
    
    // send something with a library with DerivedType as a configuration
}
```
Note that `DerivedType` and the api of the library are not under my control.
